### ISSUE TITLE
Storage Policy ID / Datastore is not applied to VM

### ISSUE THREAD

**Issue by Satak** [hashicorp/terraform-provider-vsphere#991]
### Terraform Version

`0.12.23`

### vSphere Provider Version

`1.17.0`

### Affected Resource(s)
- `vsphere_virtual_machine`

### Terraform Configuration Files

```hcl
provider "vsphere" {
  user                 = var.username
  password             = var.password
  vsphere_server       = var.server
  allow_unverified_ssl = true
}

data "vsphere_datacenter" "dc" {
  name = "datacenterName"
}

data "vsphere_resource_pool" "pool" {
  name          = "poolName"
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_network" "network" {
  name          = "networkName"
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_content_library" "library" {
  name = "libraryName"
}

data "vsphere_content_library_item" "item" {
  name       = "templateName"
  library_id = data.vsphere_content_library.library.id
}

data "vsphere_storage_policy" "policy" {
  name = "policyName"
}

resource "vsphere_virtual_machine" "vm" {
  name              = var.vmName
  resource_pool_id  = data.vsphere_resource_pool.pool.id
  storage_policy_id = data.vsphere_storage_policy.policy.id

  num_cpus = 2
  memory   = 4096
  guest_id = "windows9Server64Guest"

  network_interface {
    network_id = data.vsphere_network.network.id
  }

  disk {
    storage_policy_id = data.vsphere_storage_policy.policy.id
    label             = "disk0"
    size              = 120
  }

  clone {
    template_uuid = data.vsphere_content_library_item.item.id
    customize {
      windows_options {
        computer_name  = var.vmName
        admin_password = var.vmPassword
        workgroup      = "WORKGROUP"
        auto_logon     = true
      }
      network_interface {
        ipv4_address = var.ipAddress
        ipv4_netmask = 24
      }
      ipv4_gateway = var.gateway
    }
  }
}
```
### Error

```
Error: ServerFaultCode: The object 'vim.Datastore:' has already been deleted or has not been completely created

  on main.tf line 55, in resource "vsphere_virtual_machine" "vm":
  55: resource "vsphere_virtual_machine" "vm" { 
```

### Expected Behavior

Terraform should create VM with disk in a datastore attached to the storage policy

### Actual Behavior

Terraform creates VM with disk that is in a datastore where the template is. `storage_policy_id` doesn't affect at all to what datastore the disk is put. Even when I hardcode the storage policy id to the module level `storage_policy_id` parameter it is not working.

Noticed that the policy is set to VM and it's just the datastore that is not picked correctly from the policy.

We are using templates from the vCenter folder and not content library templates.

### Steps to Reproduce

1. Create storage policy in vCenter and add datastore to it
2. Create `main.tf` file where you clone a VM template with `storage_policy_id`
3. VM is created but not to a datastore where you have set your policy

**Comment by Satak** [comment]
This is pending support for content libraries which probably solves this issue:
https://github.com/terraform-providers/terraform-provider-vsphere/pull/985

**Comment by Satak** [comment]
@bill-rich Now the vSphere supports storage policies. Problem is that one must provide `datastore_id` or `datastore_cluster_id` stated in the documentation:

> NOTE: One of datastore_id or datastore_cluster_id must be specified.

We are using `storage_policy_id` that automatically assigns a correct datastore based on the storage policy rule so we don't want to send any datastore and just let the policy assign the datastore. Is this currently possible? Any thoughts?

**Comment by Satak** [comment]
Error when not setting any `storage_policy_id` and trying to just use `storage_policy_id`:

```
Error: ServerFaultCode: The object 'vim.Datastore:' has already been deleted or has not been completely created

  on main.tf line 55, in resource "vsphere_virtual_machine" "vm":
  55: resource "vsphere_virtual_machine" "vm" { 
```

**Comment by Satak** [comment]
@bill-rich just noticed that if you use **content library item** based VM provisioning (template uuid) your **datastore_id** that you provide to the resource is overwritten by the datastore that the template has. Is this intentional?

Other bug is that with the version **1.17.1** `guest_id` value is overwritten to empty string if you use content library item based provisioning and then your customization block will fail. This was working before. You should put an if condition before you overwrite user input variables to empty strings.

Bottom line is that currently provisioning with content library item with datastore and guest_id is broken.

**Comment by bill-rich** [comment]
This should be fixed in #1183. Please try again after the next provider release and let me know if you're still experiencing this issue.
